hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go

```diff
@@ -363,6 +378,24 @@ func reconcileAWSCluster(awsCluster *capiaws.AWSCluster, hcluster *hyperv1.Hoste
 		Host: apiEndpoint.Host,
 		Port: apiEndpoint.Port,
 	}
+
+	if hcluster.Annotations[hyperv1.AWSMachinePublicIPs] == "true" {
+		subnetIDs := sets.New[string]()
+		for i := range nodePools {
+			subnetIDPtr := nodePools[i].Spec.Platform.AWS.Subnet.ID
+			if subnetIDPtr != nil {
+				subnetIDs.Insert(*subnetIDPtr)
+			}
+		}
+		awsCluster.Spec.NetworkSpec.Subnets = nil
```

Comment:
why does awsCluster needs to know about this?

Reply:
Because machines with public IP fail to create without the subnet being specified in the AWSCluster
https://github.com/openshift/cluster-api-provider-aws/blob/6bd77fcce0c87571305edf5f1f28f1182ed25096/pkg/cloud/services/ec2/instances.go#L353-L363